demos/custom-tabs-example-app/src/main/java/org/chromium/customtabsdemos/PartialCustomTabActivity.java

```diff
@@ -0,0 +1,185 @@
+ * </ul>
+ */
+public class PartialCustomTabActivity extends AppCompatActivity implements View.OnClickListener {
+    private static final String TAG = "PartialCustomTabActivity";
+    private static final int INITIAL_HEIGHT_DEFAULT_PX = 600;
+    private static final int CORNER_RADIUS_MAX_DP = 16;
+    private static final int CORNER_RADIUS_DEFAULT_DP = CORNER_RADIUS_MAX_DP;
+    private static final int BACKGROUND_INTERACT_OFF_VALUE = 2;
+
+    private EditText mUrlEditText;
+    private CheckBox mFixedHeightCheckbox;
+    private CheckBox mBackgroundAppCheckbox;
+    private SeekBar mToolbarCornerRadiusSlider;
+    private TextView mToolbarCornerRadiusLabel;
+    private CustomTabActivityHelper mCustomTabActivityHelper;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_partial_custom);
+
+        mCustomTabActivityHelper = new CustomTabActivityHelper();
+        findViewById(R.id.start_custom_tab).setOnClickListener(this);
+
+        mUrlEditText = findViewById(R.id.url);
+        mFixedHeightCheckbox = findViewById(R.id.fixed_height);
+        mBackgroundAppCheckbox = findViewById(R.id.background_app);
+
+        mToolbarCornerRadiusLabel = findViewById(R.id.corner_radius_label);
+        mToolbarCornerRadiusSlider = findViewById(R.id.corner_radius_slider);
+        mToolbarCornerRadiusSlider.setMax(CORNER_RADIUS_MAX_DP);
+        mToolbarCornerRadiusSlider.setProgress(CORNER_RADIUS_DEFAULT_DP);
+
+        int dp = mToolbarCornerRadiusSlider.getProgress();
+        mToolbarCornerRadiusLabel.setText(getString(R.string.dp_template, dp));
+        mToolbarCornerRadiusSlider.setOnSeekBarChangeListener(new OnSeekBarChangeListener() {
+            @Override
+            public void onProgressChanged(SeekBar seekBar, int progress, boolean fromUser) {
+                String strDp = getString(R.string.dp_template, progress);
+                mToolbarCornerRadiusLabel.setText(strDp);
+            }
+
+            @Override
+            public void onStartTrackingTouch(SeekBar seekBar) {}
+
+            @Override
+            public void onStopTrackingTouch(SeekBar seekBar) {}
+        });
+    }
+
+    @Override
+    protected void onStart() {
+        super.onStart();
+        mCustomTabActivityHelper.bindCustomTabsService(this);
+    }
+
+    @Override
+    protected void onStop() {
+        super.onStop();
+        mCustomTabActivityHelper.unbindCustomTabsService(this);
+    }
+
+    @Override
+    public void onClick(View v) {
+        int viewId = v.getId();
+        switch (viewId) {
+            case R.id.start_custom_tab:
+                openCustomTab();
+                break;
+            default:
+                // Unknown View Clicked
+        }
+    }
+
+    private void openCustomTab() {
+        String url = mUrlEditText.getText().toString();
+
+        // Make sure session connection is established for partial CCT to work.
```

Comment:
Is this comment a `TODO`? You don't seem to be checking whether there's a connection.

Reply:
No, I just wanted to make it clear that CustomTabIntent builder for PCCT should make use of the established session. Put it in other way.